modules/twinkleblock.js

```diff
@@ -0,0 +1,1064 @@
+ *   forAnonOnly: <show block option in the interface only if the relevant user is an IP>
+ *   forRegisteredOnly: <show block option in the interface only if the relevant user is registered>
+ *   label: <string - label for the option of the dropdown in the interface (keep brief)>
+ *   noemail: prevent the user from sending email through Special:Emailuser
+ *   pageParam: <set if the associated block template accepts a page parameter>
+ *   prependReason: <string - prepends the value of 'reason' to the end of the existing reason, namely for when revoking talk page access>
+ *   nocreate: <block account creation from the user's IP (for anonymous users only)>
+ *   reason: <string - block rationale, as would appear in the block log,
+ *            and the edit summary for when adding block template, unless 'summary' is set>
+ *   reasonParam: <set if the associated block template accepts a reason parameter>
+ *   sig: <string - set to ~~~~ if block template does not accept "true" as the value, or set null to omit sig param altogether>
+ *   summary: <string - edit summary for when adding block template to user's talk page, if not set, 'reason' is used>
+ *   suppressArticleInSummary: <set to suppress showing the article name in the edit summary, as with attack pages>
+ *   templateName: <string - name of template to use (instead of key name), entry will be omitted from the Templates list.
+ *                  (e.g. use another template but with different block options)>
+ *   useInitialOptions: <when preset is chosen, only change given block options, leave others as they were>
+ *
+ * WARNING: 'anononly' and 'allowusertalk' are enabled by default.
+ *   To disable, set 'hardblock' and 'disabletalk', respectively
+ */
+Twinkle.block.blockPresetsInfo = {
+	'anonblock' : {
+		expiry: '31 hours',
+		forAnonOnly: true,
+		nocreate: true,
+		reason: '{{anonblock}}',
+		sig: '~~~~'
+	},
+	'anonblock - school' : {
+		expiry: '36 hours',
+		forAnonOnly: true,
+		nocreate: true,
+		reason: '{{anonblock}} <!-- Likely a school based on behavioral evidence -->',
+		templateName: 'anonblock',
+		sig: '~~~~'
+	},
+	'blocked proxy' : {
+		expiry: '1 year',
+		forAnonOnly: true,
+		nocreate: true,
+		reason: '{{blocked proxy}}',
+		sig: null
+	},
+	'blocked talk-revoked-notice' : {
+		disabletalk: true,
+		reason: 'Revoking talk page access: inappropriate use of user talk page while blocked',
+		prependReason: true,
+		summary: 'Your user talk page access has been disabled',
+		useInitialOptions: true
+	},
+	'CheckUser block' : {
+		reason: '{{CheckUser block}}',
+		sig: '~~~~'
+	},
+	'checkuserblock-account' : {
+		reason: '{{checkuserblock-account}}',
+		sig: '~~~~'
+	},
+	'checkuserblock-wide' : {
+		reason: '{{checkuserblock-wide}}',
+		sig: '~~~~'
+	},
+	'colocationwebhost' : {
+		expiry: '1 year',
+		forAnonOnly: true,
+		reason: '{{colocationwebhost}}',
+		sig: null
+	},
+	'oversightblock' : {
+		reason: '{{OversightBlock}}',
+		sig: '~~~~'
```

Comment:
These block templates serving as "templated reasons" are the only ones that don't conform to the stewardship of the user warnings WikiProject. I could have them accept `sig=yes` while also still allowing for `sig=~~~~`, but I wonder the value given all the other standard parameters aren't supported either.